### DISCUSSION TITLE
Clone instead of Copy on Radio's value?

### DISCUSSION THREAD

**Discussion by itytophile** [hecrj/iced#862] [Q&A] [answered]
Hi, I'd like my Radio button to use a String as its value but it only accepts types that implement Copy:
```rust
pub fn new<F, V>(
    value: V,
    label: impl Into<String>,
    selected: Option<V>,
    f: F
) -> Self where
    V: Eq + Copy,
    F: 'static + Fn(V) -> Message, 
```
Is there a particular reason for this design?

**Comment by hecrj** [comment, marked as answer]
The main reason is that I believe it's better to rely on `enum` values for these kinds of widgets, instead of using strings both for the value and the label.

Besides being cheaper, it should also be easier to pattern match when you get a `Message` with the selected value.

**Reply by itytophile** [reply]
I understand. In my use case, the choices are known at runtime. I could use an enum variant like `MyEnum(usize)` and look for which `String` I want in a `Vec<String>` but I decided to use buttons to directly use `String`. As I'm writing this, I think that the first solution seems to be better.

In any case, keep up the good work with Iced!
